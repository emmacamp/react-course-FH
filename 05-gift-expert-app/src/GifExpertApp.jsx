import { useState } from "react";
import { AddCategories, GiftGrid } from "./components";
// import homeSVG from './assets/aroundTheWorld.svg'

const GifExpertApp = () => {
  const [category, setCategory] = useState([]);

  const onAddCategory = (newCategory) => {
    if (category.includes(newCategory)) return;
    setCategory([newCategory, ...category])
  }

  return (
    <div className="container vh-75 justify-content-center  align-items-center">
      <div className="header p-2 justify-content-between  align-items-center">
        <h1 className="p-0 m-0">GifExpertApp</h1>
        <AddCategories addCategories={onAddCategory} stateCategories={category} />
      </div>
      <div className="container mw-100 d-flex justify-content-center  align-items-center">
        {
          category.length === 0
            ?
            <div className="d-flex flex-column w-50 h-25">
              <img className="w-100 " src={'http://perros.com/macho/doberman'} alt="Home image" />
              {/* <h1>Busca tu gif favorito</h1> */}
            </div>
            :
            category.map((ctg) => (
              <GiftGrid key={ctg} category={ctg} /> || <h1>No se encontraron datos</h1>
            ))
        }
      </div>
    </div>
  )
}



export default GifExpertApp;