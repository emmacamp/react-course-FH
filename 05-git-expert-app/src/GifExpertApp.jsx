import { useState } from "react";
import { AddCategories, GiftGrid } from "./components";


const GifExpertApp = () => {
  const [category, setCategory] = useState([]);

  const onAddCategory = (newCategory) => {
    if (category.includes(newCategory)) return;
    setCategory([newCategory, ...category])
  }

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategories addCategories={onAddCategory} stateCategories={category} />
      <div >
        {
          category.map((ctg) => (
            <GiftGrid key={ctg} category={ctg} />
          ))
        }
      </div>
    </>
  )

}



export default GifExpertApp;