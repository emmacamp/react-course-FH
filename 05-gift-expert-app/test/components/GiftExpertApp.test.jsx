import { render, screen, fireEvent } from "@testing-library/react"
import GifExpertApp from "../../src/GifExpertApp"



describe('Pruebas en <GiftExpertApp />', () => {

    test('se debe hacer el snapshot', () => {
        const { container } = render(<GifExpertApp />)
        expect(container).toMatchSnapshot()

    })

    test('Debe mostrar la imagen con el URL y el ALT indicado ', () => {
        const url = 'http://perros.com/macho/doberman'
        const title = 'Home image'

        render(<GifExpertApp />)
        // screen.debug()

        const { src, alt } = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)

    })

    test('debe haber un input y debe estar vacio', () => {
        render(<GifExpertApp />)
        screen.debug()

        const input = screen.getByRole('textbox')

        expect(input).toBeDefined()
        expect(input.value).toBe('')

    })
    test('onAddCategory debe tener una categoria', () => {
        render(<GifExpertApp />)

        const input = screen.getByRole('textbox')
        fireEvent.input(input, { target: { value: "saitama" } })
        expect(input.value).toBe("saitama");

        screen.debug()



    })

})