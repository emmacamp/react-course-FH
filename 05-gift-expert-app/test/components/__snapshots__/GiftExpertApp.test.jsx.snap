// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftExpertApp /> se debe hacer el snapshot 1`] = `
<div>
  <div
    class="container vh-75 justify-content-center  align-items-center"
  >
    <div
      class="header p-2 justify-content-between  align-items-center"
    >
      <h1
        class="p-0 m-0"
      >
        GifExpertApp
      </h1>
      <form
        aria-label="form"
      >
        <button
          type="submit"
        >
          Seach
        </button>
        <input
          placeholder="Buscar Gifts"
          type="text"
          value=""
        />
      </form>
    </div>
    <div
      class="container mw-100 d-flex justify-content-center  align-items-center"
    >
      <div
        class="d-flex flex-column w-50 h-25"
      >
        <img
          alt="Home image"
          class="w-100 "
          src="http://perros.com/macho/doberman"
        />
      </div>
    </div>
  </div>
</div>
`;
